azure-pipelines.yml

```diff
@@ -52,3 +52,12 @@ stages:
   - build
   jobs:
   - template: tools/templates/cc.yml
+- stage: e2e_tests
+  displayName: 'Triggering E2E tests'
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+  dependsOn:
+  - build
```

Comment:
E2E pipelines should be triggered only if all the above steps have passed and not only `build`.

Reply:
Added dependency on all previous stages.